src/app/parcels/ActionBar/Actions.tsx

```diff
@@ -282,18 +320,32 @@ const Actions: React.FC<Props> = ({
                 >
                     <MenuList id="action-menu">
                         {Object.entries(availableActions).map(([key, value]) => {
-                            return (
-                                <MenuItem
-                                    key={key}
-                                    onClick={onMenuItemClick(
-                                        key as ActionName,
-                                        value.errorCondition,
-                                        value.errorMessage
-                                    )}
-                                >
-                                    {key}
-                                </MenuItem>
-                            );
+                            if (key !== "Generate Map") {
```

Comment:
Could we handle this if statement withiin onMenuItemClick for clarity / consistency please? And I think AvailableActionsType could have another field like `shouldOpenModal` or something?

Now that opening the modal isn't the only type of actions, many of the existing variable names should be reviewed to avoid confusion. e.g. `selectedAction` should really be `modalTypeToDisplay` cause that's what decides whether to show modal / which modal to show